### PR TITLE
Governance candid update 2022 05 03

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Update governance candid file

# Changes

*UPDATED AUTOMATICALLY*
- governance.did
- governanceTypes.d.ts
- governance.idl.js
- governance.certified.idl.js

*error type update*
- governance.errors.ts
- governance.spec.ts

# Notes

- Tests are currently broken because of too many type differences

### CODE REVIEW

File: candid/governance.certified.idl.js [hunk 1]
```diff
@@ -70,18 +65,7 @@ export const idlFactory = ({ IDL }) => {
...
-  });
-  const KnownNeuron = IDL.Record({
-    'id' : IDL.Opt(NeuronId),
-    'known_neuron_data' : IDL.Opt(KnownNeuronData),
-  });
-  const Spawn = IDL.Record({
-    'new_controller' : IDL.Opt(IDL.Principal),
-    'nonce' : IDL.Opt(IDL.Nat64),
-  });
+  const Spawn = IDL.Record({ 'new_controller' : IDL.Opt(IDL.Principal) });
```
Comment (lmuntanerDfinity) [inline candid/governance.certified.idl.js:68]: Shouldn't this match the candid file?

```
type Spawn = record {
  percentage_to_spawn : opt nat32;
  new_controller : opt principal;
  nonce : opt nat64;
};
```

File: candid/governance.certified.idl.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Do not edit.  Compiled with ./scripts/compile-idl-js from candid/governance.did */
+/* Do not edit.  Compiled with ./scripts/compile-idl-js from candid//governance.did */
```
Comment (lmuntanerDfinity) [inline candid/governance.certified.idl.js:1]: The changes in here seem weird, there are more removed code than added, whereas in the candid file, there are more additions than removals.

And I found an example where the Record here does not match the candid file.

Could you check again?

File: candid/governance.idl.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* Do not edit.  Compiled with ./scripts/compile-idl-js from candid/governance.did */
+/* Do not edit.  Compiled with ./scripts/compile-idl-js from candid//governance.did */
```
Comment (lmuntanerDfinity) [inline candid/governance.idl.js:1]: ditto as in the certified one.

**Review by lmuntanerDfinity — COMMENTED** [review]
Some questions

**Comment by mstrasinskis** [comment]
Done in another PR https://github.com/dfinity/nns-js/pull/93

**Pull request closed without merging**